python/src/functions/create_archive.py

```diff
@@ -0,0 +1,91 @@
+import os
+import tempfile
+from typing import Any
+import zipfile
+
+import boto3
+from aws_lambda_typing.context import Context
+from mypy_boto3_s3.client import S3Client
+from src.lib.logging import get_logger, reset_contextvars
+
+S3_BUCKET = os.environ.get("S3_BUCKET", "test_bucket")
+
+
+@reset_contextvars
+def handle(event: dict[str, Any], _context: Context):
+    """Lambda handler for creating an archive of CSV files in S3
+
+    Args:
+        event: Comes from Step Function. Is a list of payloads with organization ids and reporting period ids
+        context: Lambda context
+    """
+    logger = get_logger()
+    logger.info("Received new invocation event from step function")
+    logger.info(event)
+    logger.info("Extracting payload")
+    payloads = event["Payload"]
+
+    # all payloads should have the same org_id and reporting_period_ids
+    organization_ids = {p["organization_id"] for p in payloads if p["organization_id"]}
+    reporting_period_ids = {p["reporting_period_id"] for p in payloads if p["reporting_period_id"]}
+
+    if len(organization_ids) != 1:
+        raise ValueError("All payloads must have the same organization_id")
+    if len(reporting_period_ids) != 1:
+        raise ValueError("All payloads must have the same reporting_period_id")
+
+    organization_id = organization_ids.pop()
+    reporting_period_id = reporting_period_ids.pop()
```

Comment:
Small change requested- can we update this to follow the same pattern of the other lambda functions where the payload gets validated using `pydantic` which will avoid having to manually generate these errors?

Suggested code may be like
```py
from src.lib.treasury_generation_common import OrganizationObj
from pydantic import BaseModel

class CreateArchivePayload(BaseModel):
    organization: OrganizationObj

try:
    payload = CreateArchivePayload.model_validate(event)
except...

create_archive(payload.organization.id, payload.organization.preferences.current_reporting_period_id, boto3.client("s3"), logger)
```

Example: https://github.com/usdigitalresponse/cpf-reporter/blob/main/python/src/functions/subrecipient_treasury_report_gen.py#L59-L63

Reply:
Makes sense. Code is updated to use pydantic models with test cases.